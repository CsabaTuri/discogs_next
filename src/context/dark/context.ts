import { Context as C, createContext, ProviderExoticComponent, useContext } from "react"

let initialState: any = { isDark: true, dispatch: () => { } }

const Context: C<any> = createContext<any>(initialState)

export const Provider: ProviderExoticComponent<any> = Context.Provider;

export const useDarkContext = () => useContext(Context);

export default Context

