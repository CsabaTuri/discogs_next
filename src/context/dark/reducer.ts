export function reducer(state:any, action:any) {
  switch (action.type) {
    case "isDark":
      return  {isDark: !state.isDark }
    default:
      return state
  }
}
