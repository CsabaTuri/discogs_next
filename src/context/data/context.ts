import { createContext ,useContext} from "react"

let initialState:any = {  data: {} ,dispatch: () => {}  };

 const Context = createContext(initialState)

 export const useDataContext = () => useContext(Context);

 export default Context
