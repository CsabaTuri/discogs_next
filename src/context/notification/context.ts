import { createContext, useContext } from "react";

let initialState:any = { state: { text: "", type: "success" },dispatch: () => {}  };

const Context = createContext(initialState);

export const useNotificationContext = () => useContext(Context);

export default Context;
