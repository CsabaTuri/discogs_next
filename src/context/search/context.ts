import { createContext ,useContext} from "react"

let initialState:any = { search:{search: {}},dispatch: () => {} }

 const Context = createContext(initialState)

 export const useSearchContext = () => useContext(Context);

 export default Context
