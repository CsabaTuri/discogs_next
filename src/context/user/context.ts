import { Context as C, createContext, ProviderExoticComponent, useContext } from "react"



let initialState: any = { user: {}, dispatch: () => { } };

const Context: C<any> = createContext<any>(initialState)

export const Provider: ProviderExoticComponent<any> = Context.Provider;

export const useUserContext = () => useContext(Context);

export default Context
