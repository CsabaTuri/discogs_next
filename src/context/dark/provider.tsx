import React, { useCallback, useReducer } from "react";
import { useLocalStorage } from "react-use";
import { reducer } from "./reducer";
import Context from "./context";

let initialState: any = { isDark: true, dispatch: () => { } };

const usePersistReducer = () => {
  // grab saved value from `localStorage` and
  // a function to update it. if
  // no value is retrieved, use `INITIAL_STATE`
  const [savedState, saveState] = useLocalStorage<any>("isDark", initialState);

  // wrap `reducer` with a memoized function that
  // syncs the `newState` to `localStorage` before
  // returning `newState`. memoizing is important!
  const reducerLocalStorage = useCallback(
    (state, action) => {
      const newState = reducer(state, action);

      saveState(newState);

      return newState;
    },
    [saveState]
  );

  // use wrapped reducer and the saved value from
  // `localStorage` as params to `useReducer`.
  // this will return `[state, dispatch]`
  return useReducer(reducerLocalStorage, savedState);
};

export const Provider = ({ children }: any) => {
  const [state, dispatch] = usePersistReducer();
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
};
