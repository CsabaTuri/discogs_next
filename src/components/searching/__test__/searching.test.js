import React from "react";
import { render, screen, mount } from "@testing-library/react";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Searching from "../searching";
import * as DarkContext from "../../../context/dark/context";
import "jest-styled-components";

Enzyme.configure({ adapter: new Adapter() });
const testsearchData = {
  query: "d",
  label: "d",
  year: "1",
  style: "d",
  genre: "d",
  country: "d",
  type: "master",
};
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "/",
      query: "",
      asPath: "/",
    };
  },
}));
jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));
describe("Searching", () => {
  test("snapshot test", async () => {
    await new Promise((resolve) => {
      window.setTimeout(resolve, 0);
    });

    const contextValues = { state: { isDark: false } };
    jest
      .spyOn(DarkContext, "useDarkContext")
      .mockImplementation(() => contextValues);
    const wrapper = Enzyme.mount(<Searching searchData={testsearchData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
