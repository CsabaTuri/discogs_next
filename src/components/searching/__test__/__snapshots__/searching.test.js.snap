// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searching snapshot test 1`] = `
<Searching
  searchData={
    Object {
      "country": "d",
      "genre": "d",
      "label": "d",
      "query": "d",
      "style": "d",
      "type": "master",
      "year": "1",
    }
  }
>
  <Row
    className="justify-content-md-center"
    noGutters={false}
  >
    <div
      className="justify-content-md-center row"
    >
      <Col
        md={6}
      >
        <div
          className="col-md-6"
        >
          <Form
            className="form-signin"
            inline={false}
          >
            <form
              className="form-signin"
            >
              <ForwardRef
                appear={false}
                getDimensionValue={[Function]}
                in={true}
                mountOnEnter={false}
                timeout={300}
                unmountOnExit={false}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={false}
                  aria-expanded={null}
                  enter={true}
                  exit={true}
                  in={true}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  unmountOnExit={false}
                >
                  <div
                    aria-expanded={null}
                    className="collapse show"
                    id="simple-search"
                  >
                    <FormGroup
                      controlId="artist"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="artist"
                          >
                            search
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          name="query"
                          type="text"
                          value="d"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="artist"
                            name="query"
                            type="text"
                            value="d"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </div>
                </Transition>
              </ForwardRef>
              <ForwardRef
                appear={false}
                getDimensionValue={[Function]}
                in={false}
                mountOnEnter={false}
                timeout={300}
                unmountOnExit={false}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={false}
                  aria-expanded={null}
                  enter={true}
                  exit={true}
                  in={false}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  unmountOnExit={false}
                >
                  <div
                    aria-expanded={null}
                    className="collapse"
                    id="advanced-search"
                  >
                    <FormGroup
                      controlId="artist"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="artist"
                          >
                            artist
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          name="artist"
                          type="text"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="artist"
                            name="artist"
                            type="text"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="label"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="label"
                          >
                            label
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          name="label"
                          type="text"
                          value="d"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="label"
                            name="label"
                            type="text"
                            value="d"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="year"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="year"
                          >
                            year
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          maxLength={4}
                          name="year"
                          type="text"
                          value="1"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="year"
                            maxLength={4}
                            name="year"
                            type="text"
                            value="1"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="style"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="style"
                          >
                            style
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          name="style"
                          type="text"
                          value="d"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="style"
                            name="style"
                            type="text"
                            value="d"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="genre"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="genre"
                          >
                            genre
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          name="genre"
                          type="text"
                          value="d"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="genre"
                            name="genre"
                            type="text"
                            value="d"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="type"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="type"
                          >
                            type
                          </label>
                        </FormLabel>
                        <FormControl
                          as="select"
                          name="type"
                          value="master"
                        >
                          <select
                            className="form-control"
                            id="type"
                            name="type"
                            value="master"
                          >
                            <option
                              value=""
                            >
                              all
                            </option>
                            <option
                              value="master"
                            >
                              master
                            </option>
                            <option
                              value="release"
                            >
                              release
                            </option>
                            <option
                              value="artist"
                            >
                              artist
                            </option>
                            <option
                              value="label"
                            >
                              label
                            </option>
                          </select>
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="type"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="form-label"
                            htmlFor="type"
                          >
                            country
                          </label>
                        </FormLabel>
                        <FormControl
                          autoComplete="off"
                          name="country"
                          type="text"
                          value="d"
                        >
                          <input
                            autoComplete="off"
                            className="form-control"
                            id="type"
                            name="country"
                            type="text"
                            value="d"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                  </div>
                </Transition>
              </ForwardRef>
              <Link
                href={
                  Object {
                    "pathname": "/",
                    "query": Object {
                      "country": "d",
                      "genre": "d",
                      "label": "d",
                      "query": "d",
                      "style": "d",
                      "type": "master",
                      "year": "1",
                    },
                  }
                }
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  variant="outline-info"
                >
                  <button
                    className="btn btn-outline-info"
                    disabled={false}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    type="button"
                  >
                    search
                  </button>
                </Button>
              </Link>
              <Link
                href={
                  Object {
                    "pathname": "/",
                  }
                }
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  variant="outline-info"
                >
                  <button
                    className="btn btn-outline-info"
                    disabled={false}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    type="button"
                  >
                    reset
                  </button>
                </Button>
              </Link>
              <Button
                active={false}
                aria-controls="form-collapse"
                aria-expanded={false}
                disabled={false}
                onClick={[Function]}
                variant="outline-info"
              >
                <button
                  aria-controls="form-collapse"
                  aria-expanded={false}
                  className="btn btn-outline-info"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  advanced
                </button>
              </Button>
            </form>
          </Form>
        </div>
      </Col>
    </div>
  </Row>
</Searching>
`;
